src/emeus-constraint-layout.c

```diff
@@ -1138,6 +1155,34 @@ remove_child_constraint (EmeusConstraintLayout      *layout,
   return TRUE;
 }
 
+void
+emeus_constraint_layout_activate_constraint (EmeusConstraintLayout *layout,
+                                             EmeusConstraint       *constraint)
+{
+  g_assert (constraint->solver == &layout->solver);
+
+  if (constraint->constraint == NULL)
+    return;
+
+  if (constraint->target_object == layout)
+    create_layout_constraint (layout, constraint);
+  else
+    create_child_constraint (layout, constraint->target_object, constraint);
+}
+
+void
+emeus_constraint_layout_deactivate_constraint (EmeusConstraintLayout *layout,
+                                               EmeusConstraint       *constraint)
+{
+  g_assert (constraint->solver == &layout->solver);
+
+  if (constraint->constraint == NULL)
+    return;
+
+  simplex_solver_remove_constraint (constraint->solver, constraint->constraint);
+  constraint->constraint = NULL;
```

Comment:
Let's say you deactivate a constraint with `emeus_constraint_set_active` then immediately activate it again.

Won't it hit this code path, nulling `constraint->constraint` and then in `emeus_constraint_layout_activate_constraint` hit the `if (constraint->constraint == NULL)` early return? So it won't get reactivated?

Quite possible I'm just not following here...